### DISCUSSION TITLE
How to delete an existing dataset?

### DISCUSSION THREAD

**Discussion by aghaeifar** [BlueBrain/HighFive#982] [Questions] [answered]
Hi all,

Is it feasible to replace an existing dataset with data that may have a different shape? 
I can verify the existence of the dataset name using `HighFive::File::exist(...)`, but I'm uncertain how to delete it and then create a new one using `createDataSet(...)`. 
Could you please advise on the approach to take when replacing old data is necessary? 
It's important to note that there are other datasets in the HDF5 file, so overwriting the entire file is not a viable solution.

Thanks.

**Comment by 1uc** [comment, marked as answer]
You can "delete" a dataset with:
```
grp.unlink(dset_name)
```
and create a new dataset. The issue is that unlinking will not automatically shrink the size of the file. I think it'll (sometimes) mark the space as unused and then reuse it if possible. You can use `h5repack` to copy the file, but eliminating all gaps in the original file.

Note that the special case of a growing dataset is typically solved differently. One creates a DataSet that's unlimited in the direction in which it'll grow. Then one can later resize the dataset in the unlimited direction and overwrite data as needed.
